src/maker/config.rs

```diff
@@ -49,3 +55,178 @@ impl MakerConfig {
         }
     }
 }
+
+impl MakerConfig {
+    /// new a default configuration with given port and address
+    pub fn new(file_path: Option<&PathBuf>) -> io::Result<Self> {
+        let default_config = Self::default();
+        let default_path = PathBuf::from("maker.toml");
+        let path = file_path.unwrap_or(&default_path);
+
+        // Check if the file exists
+        if !path.exists() {
+            return Err(io::Error::new(
+                ErrorKind::InvalidData,
+                format!("Config file not found: {:?}", path),
+            ));
+        }
```

Comment:
This is a hard error. This should not be the case. If file not found, use default config.

Reply:
The rusty way of handling this whole flow would be something like this..

```rust
let section = if let Some(path) = file_path {
            if path.exists() {
                parse_toml(path)?
            } else {
                log::warn!(
                    "Config file not found at path : {}, using default config",
                    path.display()
                );
                HashMap::new()
            }
        } else {
            let default_path = PathBuf::from("maker.toml");
            if default_path.exists() {
                parse_toml(&default_path)?
            } else {
                log::warn!(
                    "Config file not found in default path: {}, using default config",
                    default_path.display()
                );
                HashMap::new()
            }
        };
        ```

Reply:
@rajarshimaitra I intentionally used hard error...I thought ultimately when I gonna invoke the new() will add an `unwrap_or()` to take default in case of failure.... But I think that's not the pattern to follow..

Reply:
Okay.. That makes sense.. But better to handle it here than in higher layers of `Maker::init`..